RangePicker/tests/RangePicker-specs.js

```diff
@@ -1,105 +1,116 @@
-import {mount} from 'enzyme';
-import {RangePicker, RangePickerBase} from '../RangePicker';
+import '@testing-library/jest-dom';
+import {render} from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
 
-const tap = (node) => {
-	node.simulate('mousedown');
-	node.simulate('mouseup');
-};
-const decrement = (slider) => tap(slider.find('Button').last());
-const increment = (slider) => tap(slider.find('Button').first());
+import {RangePicker, RangePickerBase} from '../RangePicker';
 
 describe('RangePicker Specs', () => {
 	test('should render a single child with the current value', () => {
-		const picker = mount(
+		const {getByText} = render(
 			<RangePicker min={-10} max={20} value={10} />
 		);
 
-		const expected = '10';
-		const actual = picker.find('PickerItem').text();
+		const value = getByText('10');
+		const expected = 'item';
+		const actual = value.className;
 
-		expect(actual).toBe(expected);
+		expect(value).toBeInTheDocument();
+		expect(actual).toContain(expected);
 	});
 
 	test('should increase by step amount on increment press', () => {
-		const picker = mount(
+		const {getByText} = render(
 			<RangePicker min={0} max={100} defaultValue={10} step={1} noAnimation />
 		);
 
-		increment(picker);
+		userEvent.click(getByText('▶'));
```

Comment:
I have a question because I am not familiar with React testing library.
What is the difference between fireEvent and userEvent?

Reply:
Yes. userEvent uses fireEvent but also it starts a chain of actions based on the element you are trying to click.

Reply:
Sorry. I want to know a little more. In this case, what happens when you use fireEvent.click?

Reply:
In this particular case it will not work. You need to be very specific what you target with fireEvent. It is recommanded also that we use userEvent because it is closer to the user's input. Most of the time fireEvent will be used when there is not a direct function that replaces it in userEvent(ex: fireEvent.mouseDown or fireEvent.keyDown because it simulates just the initial part of a click or keypress).

Reply:
thank you I understand.